app/controllers/application_controller.rb

```diff
@@ -5,6 +5,173 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    #added below
+    enable :sessions
+    set :session_secret, "hashtagwars"
   end
 
-end
\ No newline at end of file
+  get '/' do 
+    erb :index
+  end
+
+  get '/login' do 
+    if is_logged_in?
+      redirect '/tweets'
+    else
+      erb :"users/login"
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by_username(params[:username])
+    session[:id] = @user[:id]
+    if @user.password == params[:password]
```

Comment:
You should be using [`has_secure_password`](http://api.rubyonrails.org/classes/ActiveModel/SecurePassword/ClassMethods.html#method-i-has_secure_password) to give user's passwords and be able to authenticate against them.